### PR TITLE
Skysim5000

### PR DESCRIPTION
**Pull request by evevkovacs**
> If this pull request is to address specific issues, please add "fix #ISSUE_NUMBER" below so that when this PR is merged, the associated issues will be automatically closed. Always use a short but descriptive title; avoid using "issues/xxx" as title.

Update reader to handle bad magnification values and add configuration files for skysim5000_v1.1.1 and for the test catalog skysim5000_v2.0.1_small_colortest.  The latter catalog is for testing the color-shift approach to mitigating discreteness effects at high redshift in cosmDC2.

### CODE REVIEW

**Review by yymao — CHANGES_REQUESTED** [review]
Thanks, @evevkovacs! I'm really excited to see Skysim5000 getting ready! I have one question about the reader and one about the release plan:

1. Since Skysim5000 shares the cosmoDC2 reader, the new updates to the calculation of magnification and lensed magnitude will affect cosmoDC2 catalogs, resulting in different values when people load cosmoDC2, even if the cosmoDC2 version has not changed. I think we have to make the quantity modifiers backward compatible by adding some version conditions; see examples [here](https://github.com/LSSTDESC/gcr-catalogs/blob/skysim5000/GCRCatalogs/cosmodc2.py#L513). 

2. As we discussed (also with @JoanneBogart), releasing Skysim5000 in its current HDF5 format would make interactive use practically difficult to due its large size. So I wonder if we should prepare the parquet version before advertising and encouraging end users to work with Skysim5000?

**Comment by evevkovacs** [comment]
@yymao Thanks Yao. I thought about both of these issues before submitting the PR.
1. The change to the magnification function that would affect cosmoDC2 is for magnification values that are negative or > 2. I believe that neither of these cases occurred for cosmoDC2, but I have to check to make sure that there are no values >2.  In this case the problem is moot.  However, even if there are a few values > 2, I would argue that this provision should have been in the reader in the first place, since we are saying that any values >2 are not to be trusted. So I think of it as a bug fix.
2. I think that it is worth releasing the catalog and warn people that accessing the full version will take a long time. The CL group (for example) has been waiting a while, and they may want to go through it and pull off a sub-sample of clusters.  Other TJP projects could also get going with an immediate release.  So I come down on the side of releasing it sooner rather than later.

**Comment by yymao** [comment]
@evevkovacs I agree that we can treat the edit to magnification calculation as a bug fix to cosmoDC2, but I think it should has a new version number, and not be implemented on cosmoDC2 v1.1.4. While the number of objects affected may be negotiable, I think silently updating values in the widely used cosmoDC2 v1.1.4 catalog without a version increment is likely to result in future confusion.

**Comment by evevkovacs** [comment]
@yymao  I just pushed the new version of the reader. I tested it on the small catalog and got no errors. I think it addresses the magnification issues you raised. A few people in WL are using the reader to run DESCQA tests so I think it would be good to have it generally available. Thanks

File: GCRCatalogs/catalog_configs/skysim5000_v1.0.0_image.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#  Use ^/ to indicate file path relative to GCR root dir
+
+subclass_name: cosmodc2.CosmoDC2GalaxyCatalog
```
Comment (yymao) [inline GCRCatalogs/catalog_configs/skysim5000_v1.0.0_image.yaml:3]: Should this be updated to the new class?

**Review by yymao — CHANGES_REQUESTED** [review]
Thanks, @evevkovacs. One question --- do you intend to add skysim5000 v1.0.0 to GCRCatalogs, or just v1.1.1 and v2.0.1? 

If you do, then should v1.0.0 use the new reader class or the original cosmoDC2 class?

Some minor code format issues are left inline.

File: GCRCatalogs/catalog_configs/skysim5000_v1.0.0_small.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    H0: 71.0
+    Om0: 0.2648
+    Ob0: 0.0448
+    sigma8: 0.8
+    n_s: 0.963
+
+lightcone: true
+
+version: "1.0.0"
+
```
[the remark this answers is not in the document]
Comment (evevkovacs) [reply GCRCatalogs/catalog_configs/skysim5000_v1.0.0_small.yaml:18]: @yymao Oops, sorry about this. I had forgotten that I pushed these to the repository. I don't think that these should go into the master branch. The magnifications in v1.0.0 are wrong and the only reason we are keeping it around in the CS area is for comparison with the color discreteness tests we have been doing. Can all these v1.0.0  config files be deleted from the PR?.  (I'm not sure how to do that)

[the remark this answers is not in the document]
Comment (evevkovacs) [reply GCRCatalogs/catalog_configs/skysim5000_v1.0.0_small.yaml:18]: @yymao To clarify,  I want to keep them in my branch for the time being, but not in the PR.

[the remark this answers is not in the document]
Comment (yymao) [reply GCRCatalogs/catalog_configs/skysim5000_v1.0.0_small.yaml:18]: @evevkovacs Yes, it's possible. Follow the steps here:

```sh
git checkout skysim5000
git pull -r
git rm --cached GCRCatalogs/catalog_configs/skysim5000_v1.0.0_*
git commit -m "remove v1.0.0"
git push origin skysim5000
```

[the remark this answers is not in the document]
Comment (evevkovacs) [reply GCRCatalogs/catalog_configs/skysim5000_v1.0.0_small.yaml]: @yymao Thanks a lot. skysim5000_v1.0.0_* is now removed from the PR. I think that means it is ready to merge.

**Review by yymao — APPROVED** [review]
Thanks, @evevkovacs

**Pull request merged**